Week 04/Lecture 08/Assignment 03/src/main/java/com/lecture8/EmployeeCRUD/dao/EmployeeDAO.java

```diff
@@ -0,0 +1,109 @@
+package com.lecture8.EmployeeCRUD.dao;
+
+import com.lecture8.EmployeeCRUD.models.Employee;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Repository
+public class EmployeeDAO {
+
+    @Autowired
+    @Qualifier("jdbcTemplate1")
+    private JdbcTemplate jdbcTemplate1;
+
+    @Autowired
+    @Qualifier("jdbcTemplate2")
+    private JdbcTemplate jdbcTemplate2;
+
+    public List<Employee> findAll() {
+        String sql = "SELECT * FROM employees";
+        return jdbcTemplate1.query(sql, new BeanPropertyRowMapper<>(Employee.class));
+    }
+
+    public Employee findById(String id) {
+        String sql = "SELECT * FROM employees WHERE id = ?";
+        return jdbcTemplate1.queryForObject(sql, new Object[]{id}, new BeanPropertyRowMapper<>(Employee.class));
+    }
+
+    public List<Employee> findByDepartment(String department) {
+        String sql = "SELECT * FROM employees WHERE department = ?";
+        return jdbcTemplate1.query(sql, new Object[]{department}, new BeanPropertyRowMapper<>(Employee.class));
+    }
+
+    @Transactional(transactionManager = "transactionManager1")

```

Comment:
Question: what are case @Transactional does not work fine. Pls explain.

Reply:
Yes, that's one of the case. Another case is for private method - should notice that.